### ISSUE TITLE
Global Search

### ISSUE THREAD

**Issue by luisgmgouveia** [FSHSchool/fsh-finder#12]
Fist of all thank you very much for this project. It's amazing.

This is not really an issue, more like an improvement proposal:

**- It would be amazing if we could have a global search to crawl inside the (almost) 500 repositories.**

This morning I searched for: "->" path:*.fsh

On the 11th repository (11th manual search) I ended up finding the example I was searching for, but it could have been so much quicker...

Thanks!

**Comment by cmoesel** [comment]
Thanks for the suggestion. @masnick was the primary developer of this tool -- so he'd be the best person to determine the feasibility of such a feature. That said, I suspect it might be difficult, as FSH Finder would need to maintain an index of all the repositories' contents in order to support search. This index would likely be large, and since FSH Finder works as a static page (with no smart backend server), this large index would need to be downloaded to the browser as part of the web page. But I'll let Max chime in on this!

**Comment by luisgmgouveia** [comment]
Cool! Thanks!

**Comment by masnick** [comment]
Chris is right that arbitrary search would not be possible with the current architecture.

But one thing we could do is add searches for use of specific FSH code to the script that generates the static site. For example, we could have a flag for all the `.fsh` files that contain `->`. The trick here is generating the list searches that would be broadly useful. I'm not sure what the best approach would be here, or if it would even be useful (arguably a lot of this kind of search is for something that isn't broadly used.)

Another option would be to quantify how "big" a given project is. That way you could narrow your manual searching down to repos that have the most FSH, which would presumably increase the likelihood of finding an example of the thing you're looking for.

So no easy answer, but there are some options here that may at least partially address this use case.

**Comment by luisgmgouveia** [comment]
I agree that quantifying the projects in the list, with an additional column, would be very helpful. I can confirm this because I kept track, manually, of the projects that had more examples.
